---
test-name: CommandInvokerResponseIndicatesInvocationError_ThrowsException
description:
  condition: >-
    CommandInvoker receives response message with status indicating the request data is not valid at the application level.
  expect: >-
    Invocation throws 'invocation error' exception.

prologue:
  invokers:
  - { }

actions:
- action: invoke command
  invocation-index: 0
- action: await publish
  correlation-index: 0
- action: receive response
  correlation-index: 0
  status: "422" # Unprocessable Content
  is-application-error: "true"
  metadata:
    "__stMsg": "This is a content error with details"
    "__propName": "requestHeader"
    "__propVal": "requestValue"
- action: await invocation
  invocation-index: 0
  catch:
    error-kind: invocation error
    in-application: !!bool true
    is-shallow: !!bool false
    is-remote: !!bool true
    status-code: 422
    message: "This is a content error with details"
    supplemental:
      property-name: 'requestheader'
      property-value: "requestValue"
...
